src/main/lrsql/system/lrs.clj

```diff
@@ -136,8 +142,11 @@
                     stmt-result)
                 ;; Non-error result - continue
                 (if-some [stmt-id (:statement-id stmt-result)]
-                  (recur (rest stmt-ins)
-                         (update stmt-res :statement-ids conj stmt-id))
+                  (do
+                    ;; Submit statement for reaction if enabled
+                    (react-init/offer-trigger! reaction-channel stmt-id)
+                    (recur (rest stmt-ins)
```

Comment:
what is `rest` doing in this case? Ensuring that insert has completed? Is it lazy?

Reply:
That's part of the existing code here, it loops through the sequence of statements, storing them and keeping the ids for novel ones